src/Libraries/CoreNodeModels/Logic/If.cs

```diff
@@ -72,4 +74,44 @@ public override IEnumerable<AssociativeNode> BuildOutputAst(List<AssociativeNode
             };
         }
     }
+
+    [NodeName("NewIf")]
```

Comment:
Why do we need to rename it here?

Reply:
We still need to decide on the final name for this node. I have just added this name as a draft. Since, the old "IF" node still exists in the Dynamo library, we will need a different name for the node.

Reply:
I mean to ask if we can keep the same name for the new node while deprecating the old one.

Reply:
@reddyashish @QilongTang I'm still not sure about this. To give an example, I introduced a new `IntegerSlider64Bit` `NodeModel` class but could keep the same name `Integer Slider` for the node. I think we may be able to do something similar here so that new `If` nodes have new behaviour but the same name. We might just have to figure out a way for old `If` nodes to continue to deserialize as the old NodeModel class.

Reply:
@aparajit-pratap what about if the old IF node actually has its name migrated to "replicating if" or "legacy if" and the new node becomes "If"?

Reply:
@aparajit-pratap we can have both the nodes with the same name but I feel it might cause some confusion to the users who search for that node as both will be visible now. I feel we should have a different name till we completely obsolete the old node. @mjkkirschner's suggestion looks good to me. What do you think?

Reply:
@reddyashish I think you're getting confused. The old node will be hidden from the library.

Reply:
@reddyashish - AFAIK @aparajit-pratap is correct that the use of the `[obsolete]` attribute should hide the node - it's possible you need to use `[NodeObsolete]` for that though - I know there is some discrepancy between zero touch and nodeModel nodes though - so watch out for that.

Maybe a good first step is to decide what should happen?

1. should the original node be hidden, but still load when in an existing graph to stay backward compatible?
2. If a user places a new node what should it be called?
3. Is there any way to tell the difference between the old and new nodes in the same graph? (I assume the old ones will be in error state because of the obsolete attr - if it's working right?)

Reply:
IMHO, from a user perspective they want the following

> @reddyashish - AFAIK @aparajit-pratap is correct that the use of the `[obsolete]` attribute should hide the node - it's possible you need to use `[NodeObsolete]` for that though - I know there is some discrepancy between zero touch and nodeModel nodes though - so watch out for that.
> 
> Maybe a good first step is to decide what should happen?
> 
> 1. should the original node be hidden, but still load when in an existing graph to stay backward compatible?
> 2. If a user places a new node what should it be called?
> 3. Is there any way to tell the difference between the old and new nodes in the same graph? (I assume the old ones will be in error state because of the obsolete attr - if it's working right?)

@mjkkirschner IMHO, from a user perspective:
1. Yes this should happen
2. It should simply be called "If" but reference the updated, new node. 
3. Is the error message obvious if we do this? Or will it simply say it's missing something? Ideally, users would understand that they need to replace that node. Would Copy + Paste of that node into the same place do the trick?

Reply:
I have updated the code to hide the old "If" node from the library and rename the new one to have the same name. When the users open an old graph with the if node, it will behave like the old one(conditional if). Any new "If" node that will be placed will build on the new functionality. The [NodeObsolete] attribute can only be used for a method or a property and the [obsolete] attribute will throw a warning when the old "if" class is used in any of the code. It wouldn't throw an error on the Dynamo side. Do we want to throw an error or warning for the old "if" nodes?

Reply:
After discussing with Sol, we will be showing a warning message on the "If" node when an old graph is opened..

<img width="418" alt="Screen Shot 2021-02-16 at 10 52 22 AM" src="https://user-images.githubusercontent.com/43763136/108087571-92a0d400-7045-11eb-89ca-c0e0e4647f0b.png">